### PR TITLE
Updated API for in-place filtering, updated examples, updated README.

### PR DESCRIPTION
**Pull request by virtualritz**

### CODE REVIEW

**Comment by Twinklebear** [comment]
Just a heads up, I haven't forgotten about this but have been tied up with some other projects and other external stuff

File: examples/denoise_exr/src/main.rs [hunk 1]
```diff
@@ -106,55 +108,54 @@ fn main() {
...
-                .execute_with_albedo_normal(
-                    &color.img[..],
-                    &albedo.img[..],
-                    &normal.img[..],
-                    &mut filter_output[..],
-                )
-                .expect("Invalid input image dimensions?");
+
+    #[allow(unused_assignments)]
+    let mut albedo = EXRData::new(0, 0);
```
Comment (Twinklebear) [inline examples/denoise_exr/src/main.rs]: For this and `normal` I think we could make them `let mut albedo: EXRData` ?

File: README.md [hunk 2]
```diff
@@ -26,11 +26,12 @@ fn main() {
...
     let device = oidn::Device::new();
-    let mut filter = oidn::RayTracing::new(&device);
-    // Optionally add float3 normal and albedo buffers as well
-    filter.set_srgb(true)
-        .set_img_dims(input.width() as usize, input.height() as usize);
-    filter.execute(&input_img[..], &mut filter_output[..]).expect("Filter config error!");
+    oidn::RayTracing::new(&device)
+        // Optionally add float3 normal and albedo buffers as well
+        .srgb(true)
+        .set_image_dimensions(input.width() as usize, input.height() as usize);
```
Comment (Twinklebear) [inline README.md]: Should be `image_dimensions`? based on the change to the filter API

File: src/filter.rs [hunk 3]
```diff
@@ -13,6 +13,8 @@ pub enum FilterError {
 pub struct RayTracing<'a> {
     handle: OIDNFilter,
     device: &'a Device,
+    albedo: Option<&'a [f32]>,
+    normal: Option<&'a [f32]>,
```
Comment (Twinklebear) [inline src/filter.rs]: So maybe using the same lifetime as the filter is ok for https://github.com/Twinklebear/oidn-rs/issues/5 ? I'm not sure if that introduces a different issue with the device and albedo/normal being marked as having the same life times?

E.g. does some code like the below compile? I forget a bit the bounding/rules of lifetimes now actually, whether marking them all as `'a` is saying none should outlive the device, or tying them to a similar lifetime as the device.

```
let device = oidn::Device::new();
{
    let mut filter = oidn::RayTracing::new(&device);
    let albedo = load_exr(&albedo_exr);
    filter
        .set_srgb(true)
        .set_img_dims(input.width() as usize, input.height() as usize)
        .albedo(&albedo.img[..])
       ... /// etc.
}
```

**Review by Twinklebear — CHANGES_REQUESTED** [review]
For the conflict, I guess it's on us both changing the `&b"Blah"` to `b"Blah"`? Are there some other conflicts between this and master?

**Comment by Twinklebear** [comment]
Let me also try to get travis running for the PRs, I'm not sure why it didn't run here

**Edit** Nevermind, that's due to the merge conflict. So once it becomes automatically mergeable travis will run

File: src/filter.rs (hunk 3 above)
Comment (virtualritz) [reply src/filter.rs]: I added a 2nd lifetime for `albedo` & `normal`. New PR incoming. :)

File: examples/denoise_exr/src/main.rs (hunk 1 above)
Comment (virtualritz) [reply examples/denoise_exr/src/main.rs]: One can actually do `let albedo: EXRData;` I didn't know this at all. I.e. that Rust knows forward declarations to help resolve lifetime issues w/o introducing a `mut` requirement. Neat.

File: README.md (hunk 2 above)
Comment (virtualritz) [reply README.md]: Yes, I fixed the code in the README.

**Pull request closed without merging**